app/assets/javascripts/discourse/app/lib/uploads.js

```diff
@@ -29,30 +29,28 @@ export function validateUploadedFiles(files, opts) {
   }
 
   const upload = files[0];
+  return validateUploadedFile(upload, opts);
+}
 
+export function validateUploadedFile(file, opts) {
   // CHROME ONLY: if the image was pasted, sets its name to a default one
   if (typeof Blob !== "undefined" && typeof File !== "undefined") {
```

Comment:
I moved these lines and the type guesser to validateUploadedFile because it is acting on a single file...I think in future we just get rid of validateUploadedFiles because it only operates on the first file in the array too (this looks to be because of how jquery file uploader structures its files)